docs/content/querying/sql.md

```diff
@@ -1,77 +1,328 @@
+Each Druid datasource appears as a table in the "druid" schema. This is also the default schema, so Druid datasources
+can be referenced as either `druid.dataSourceName` or simply `dataSourceName`.
+
+Identifiers like datasource and column names can optionally be quoted using double quotes. To escape a double quote
+inside an identifier, use another double quote, like `"My ""very own"" identifier"`. All identifiers are case-sensitive
+and no implicit case conversions are performed.
+
+Literal strings should be quoted with single quotes, like `'foo'`. Literal strings with Unicode escapes can be written
+like `U&'fo\00F6'`, where character codes in hex are prefixed by a backslash. Literal numbers can be written in forms
+like `100` (denoting an integer), `100.0` (denoting a floating point value), or `1.0e5` (scientific notation). Literal
+timestamps can be written like `TIMESTAMP '2000-01-01 00:00:00'`. Literal intervals, used for time arithmetic, can be
+written like `INTERVAL '1' HOUR`, `INTERVAL '1 02:03' DAY TO MINUTE`, `INTERVAL '1-2' YEAR TO MONTH`, and so on.
+
+Druid SQL supports SELECT queries with the following structure:
 
 ```
-jdbc:avatica:remote:url=http://BROKER:8082/druid/v2/sql/avatica/
+[ EXPLAIN PLAN FOR ]
+[ WITH tableName [ ( column1, column2, ... ) ] AS ( query ) ]
+SELECT [ ALL | DISTINCT ] { * | exprs }
+FROM table
+[ WHERE expr ]
+[ GROUP BY exprs ]
+[ HAVING expr ]
+[ ORDER BY expr [ ASC | DESC ], expr [ ASC | DESC ], ... ]
+[ LIMIT limit ]
 ```
 
-Example code:
+The FROM clause refers to either a Druid datasource, like `druid.foo`, an [INFORMATION_SCHEMA table](#retrieving-metadata), a
+subquery, or a common-table-expression provided in the WITH clause. If the FROM clause references a subquery or a
+common-table-expression, and both levels of queries are aggregations and they cannot be combined into a single level of
+aggregation, the overall query will be executed as a [nested GroupBy](groupbyquery.html#nested-groupbys).
+
+The WHERE clause refers to columns in the FROM table, and will be translated to [native filters](filters.html). The
+WHERE clause can also reference a subquery, like `WHERE col1 IN (SELECT foo FROM ...)`. Queries like this are executed
+as [semi-joins](#query-execution), described below.
+
+The GROUP BY clause refers to columns in the FROM table. Using GROUP BY, DISTINCT, or any aggregation functions will
+trigger an aggregation query using one of Druid's [three native aggregation query types](#query-execution).
+
+The HAVING clause refers to columns that are present after execution of GROUP BY. It can be used to filter on either
+grouping expressions or aggregated values. It can only be used together with GROUP BY.
+
+The ORDER BY clause refers to columns that are present after execution of GROUP BY. It can be used to order the results
+based on either grouping expressions or aggregated values. It can only be used together with GROUP BY.
```

Comment:
I'm not aware of any plans to support ORDER BY for non-aggregation queries, but it could be done in principle.